pyHS100/discover.py

```diff
@@ -53,7 +54,11 @@ def discover(protocol: TPLinkSmartHomeProtocol = None,
                 ip, port = addr
                 info = json.loads(protocol.decrypt(data))
                 device_class = Discover._get_device_class(info)
-                if device_class is not None:
+                if deviceType is not None:
```

Comment:
```suggestion
                if device_type is not None and device_class is not None:
```
to avoid another layer of nesting below. See my comment in the PR though, I think it will be much better when the type is defined as an enum.